### PR TITLE
Use priority tags when sorting puzzles

### PR DESCRIPTION
**Pull request by leonyz**
Puzzles are sorted by priority tag (high priority > no tag > low priority). Sort is within puzzle type, so low-priority feeders still appear above all metas.

### CODE REVIEW

File: hunts/src/puzzlesSlice.js [hunk 1]
```diff
@@ -96,6 +96,30 @@ function puzzleComparator(a, b) {
   } else if (a.is_meta && !b.is_meta) {
     return 1;
   }
+  // High-priority first
```
Comment (asdfryan) [inline hunts/src/puzzlesSlice.js:99]: If it were me, I would:

1) Write a helper function `priority(row)` that returns 1 if hi-pri, else if-1 if lo-pri, else 0.
2) Return `priority(a) - priority(b)` if nonzero. Else, fall back to b.id - a.id;

Comment (leonyz) [reply hunts/src/puzzlesSlice.js]: Latest commit follows your suggestion.

File: hunts/src/puzzlesSlice.js [hunk 2]
```diff
@@ -83,6 +83,15 @@ export const addPuzzleTag = createAsyncThunk(
   }
 );
 
+function priority(row) {
```
Comment (asdfryan) [inline hunts/src/puzzlesSlice.js:86]: Can you scope this inside puzzleComparator()?

Also, for consistency could you you return 1 for Hi-pri and -1 for low-pri, and return `priority(b)-priority(a)`?

**Review by asdfryan — APPROVED** [review]

**Comment by rgossiaux** [comment]
Thank you!!

**Pull request merged**